### PR TITLE
Fix rolling update --image 

### PR DESCRIPTION
**Pull request by bprashanth**
Creating an rc and trying to perform a rolling update with --image will fail consistently with a resource version error because we update the existing rc twice without updating the rv inbetween. This PR updates the in memory copy of the existing rc and adds a test for rolling-update.

### CODE REVIEW

**Comment by bprashanth** [comment]
@brendandburns @jlowdermilk

File: hack/test-cmd.sh [hunk 1]
```diff
@@ -541,6 +545,14 @@ __EOF__
   kubectl delete pod valid-pod "${kube_flags[@]}"
   kubectl delete service frontend{,-2,-3} "${kube_flags[@]}"
 
+  ### Perform a rolling update with --image
+  # Pre-condition status.Replicas is 3, otherwise the rcmanager could update it and interfere with the rolling update
+  kube::test::get_object_assert 'rc frontend' "{{$rc_status_replicas_field}}" '3'
+  # Command
+  kubectl rolling-update frontend --image=kubernetes/pause --update-period=10ns --poll-interval=10ms "${kube_flags[@]}"
+  # Post-condition: current image IS kubernetes/pause
+  kube::test::get_object_assert 'rc frontend' '{{range \$c:=$rc_container_image_field}} {{\$c.image}} {{end}}' ' +kubernetes/pause +'
```
Comment (bprashanth) [inline hack/test-cmd.sh]: I get the feeling there's a cleaner way to do this, still digging through the go templating api. Suggestions?

**Comment by bprashanth** [comment]
As a spinoff question, I also noticed we leave the deployment label on the new rc, should we remove it when we rename it?

File: hack/test-cmd.sh (hunk 1 above)
Comment (jlowdermilk) [inline hack/test-cmd.sh]: That looks about as clean as you're likely to get it. go text/template is not very pretty.

**Comment by jlowdermilk** [comment]
LGTM, but you'll need to rebase.

As for the deployment label question, in theory we could remove the deployment label when we rename it, but I think it's actually a good idea to keep it; it does no harm and keeping it somewhat simplifies subsequent rolling updates. I don't know whether keeping it was intentional or just convenient/harmless.

**Comment by cjcullen** [comment]
#7570 seems closely related

**Comment by alindeman** [comment]
> #7570 seems closely related

Yah, it addresses the same issue. I hadn't seen this PR before I made mine. It seems like this one might be superior since it contains a test and handles a few other cases I didn't consider.

**Comment by bprashanth** [comment]
I guess this is ready for merge if @jlowdermilk is Ok with it and it passes travis/shippable. Apologies for the duplication @alindeman!

**Comment by jlowdermilk** [comment]
lgtm will merge on green.

**Comment by alindeman** [comment]
> I guess this is ready for merge if @jlowdermilk is Ok with it and it passes travis/shippable. Apologies for the duplication @alindeman!

No problem at all! It took me more time to find the issue than to fix it, so I didn't know what to search for until very recently :) Will be glad to have it fixed in either case.

**Comment by jlowdermilk** [comment]
looks like you need a rebase

**Comment by bprashanth** [comment]
This file changes faster than i can run make release =\
Rebased

**Comment by bprashanth** [comment]
Tracking status.replicas on the rc is not a smart thing, it'll work 90% of the time but status.replicas is kind of best effort. The better fix would be to put the two update(rc) calls in rollingupdate in a retry loop and remove the status.replicas assertion from the test.

**Comment by bprashanth** [comment]
@jlowdermilk Heh, I was actually going to modify the pr to do as my last comment suggested. No problem, I'll send a follow up and cross my fingers that the new status.replicas check in test-cmd doesn't flake too much.

**Comment by jlowdermilk** [comment]
Yeah follow up is fine. Fixed command with flaky test is better than permanently broken command with no test.

**Comment by brendandburns** [comment]
@bprashanth Thanks for fixing my bugs (and adding a test I should have added in the first place)

**Pull request merged**